packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -55,25 +88,64 @@ export default ({ strapi }: { strapi: Strapi }) => ({
     }) as unknown as DocumentMetadata['availableStatus'][0] | null;
   },
 
-  // TODO: Modified status
-  async getStatus(uid: Common.UID.ContentType, document: DocumentVersionSelector) {
-    if (document.publishedAt) return 'published';
-    return 'draft';
+  /**
+   * Get the available status of many documents, useful for batch operations
+   * @param uid
+   * @param documents
+   * @returns
+   */
+  async getManyAvailableStatus(uid: Common.UID.ContentType, documents: DocumentVersionSelector[]) {
+    if (!documents.length) return [];
+
+    // The status of all documents should be the same
+    const status = documents[0].publishedAt !== null ? 'published' : 'draft';
+    const otherStatus = status === 'published' ? 'draft' : 'published';
+
+    return strapi.documents(uid).findMany({
+      filters: {
+        id: { $in: documents.map((d) => d.id) },
+      },
+      status: otherStatus,
+      fields: ['id', 'locale', 'updatedAt', 'createdAt', 'publishedAt'],
+    }) as unknown as DocumentMetadata['availableStatus'];
+  },
+
+  getStatus(
+    document: DocumentVersionSelector,
+    otherStatusDocument?: DocumentMetadata['availableStatus']
+  ) {
+    const isPublished = document.publishedAt !== null;
+
+    // Status should be published when returning the published data
+    if (isPublished) {
+      return CONTENT_MANAGER_STATUS.PUBLISHED;
+    }
+
+    // Document is a draft version
+    const publishedVersion = otherStatusDocument?.find((d) => d.publishedAt !== null);
+
+    // There is only a draft version
+    if (!publishedVersion) {
+      return CONTENT_MANAGER_STATUS.DRAFT;
+    }
+
+    // The draft version is the same as the published version
+    if (publishedVersion && areDatesEqual(document.updatedAt, publishedVersion.updatedAt, 500)) {
```

Comment:
Will the dates/times actually differ in ms in the DB? If so I wonder if we should find a way to make sure the updatedAt times are the same? the threshold approach doesn't seem ideal

Reply:
yep! I added a todo in the areDatesEqual function, 
I will be setting the updated at value to be the same in both draft and published versions. ATM I didn't find another way than doing this

Reply:
Can we just do this given the check abpove?
```suggestion
    if (areDatesEqual(document.updatedAt, publishedVersion.updatedAt, 500)) {
```